### DISCUSSION TITLE
Improve and talk more about Grails scaffolding features

### DISCUSSION THREAD

**Discussion by kico-itexto** [apache/grails-core#13533] [Ideas]
Grails has the best scaffold I had ever used. It's great, easy to customize and, more important: generate GREAT code.

It was a shame when at the time of the release of Grails 3 there was plans to remove this feature from Grails (thank you for didn't doing that!)

Now we are seeing more and more people using SSR in frameworks like Angular, Vue, React. I think that Grails should invest more on this feature. But more than invest and improve, show to the world what we can do with it.

Because as a consultant, many times the main feature of Grails that customers wanted and LOVED was the scaffolding mechanism. And even today, this is one of the main differentials of the framework.

The closest we get is the admin module of Django (which have many limitations). jHipster is not even close because it don't deal well with legacy databases and Ruby on Rails is waaaaaaaay behind Grails in this feature.

**Comment by gsartori** [comment]
It's not a Scaffolding feature but I kind of feel you may be interested in Dueuno Elements: https://medium.com/@gianlucasartori/my-first-dueuno-elements-application-in-3-steps-tutorial-4210e606c204

**Comment by jamesfredley** [comment]
I agree with this comment for several parts of Grails.  The fields plugin, which scaffolding leverages, is the best form field rendering solution I have seen.  It is the DRYest and a pleasure to use.  

Enhancing the documentation and having more examples will definitely spread the word.

**Reply by chrismalan** [reply]
Amen. Enhancing the Fields plugin documentation and supplying many examples is sorely needed.

**Reply by jamesfredley** [reply]
https://grails.apache.org/guides/grails-fields-custom-widgets-and-wrappers/8/guide/index.html

**Reply by chrismalan** [reply]
Looked at the page, downloaded the code and tried to run it. First, there's no Date.parse(String, String) method for loading the database in Bootstrap.groovy. Fixed that and it compiled and ran. Book.list works fine, but Author list throws an exception. It's looking for a 'title' property in Author, which doesn't exist. 
_org.grails.taglib.GrailsTagException: [views/author/index.gsp:14] Error executing tag <f:table>: [views/templates/_fields/_table.gsp:28] Error executing tag <g:form>: Error evaluating expression [book.title.encodeAsJavaScript()] on line [25]: No such property: title for class: example.Author_
Clearly, they didn't try to run their own code. Not the way to go.

**Reply by jamesfredley** [reply]
The bug was introduced by a fix. Commit 494bc4d47c ("correct table-template paths", an ancestor of master) correctly found that the original snippet sat at the unreachable path _fields/book/_table.gsp and moved it to the working app-wide path templates/_fields/_table.gsp when adding test coverage to the 10 new Grails 8 guides.

Another update will fix this new issue.

**Reply by jamesfredley** [reply]
the Grails Guides have been migrated over to the official Apache Grails site → https://grails.apache.org/guides/

Even better news: the process for adding new guides has been completely overhauled and is now ridiculously simple and fast.

Guides have always been one of the most helpful resources for learning Grails - whether you’re brand new or picking up advanced techniques. We really want to bring this section back to life and get fresh, useful guides flowing again on all kinds of topics.

:memo: Full details on the new easy process (check the README):
https://github.com/apache/grails-static-website#authoring-a-guide

And here’s the first new guide created with the new workflow:
Grails Fields Plugin: Custom Widgets, Wrappers, and Tables
A really thorough, practical guide on using the Fields plugin to build clean CRUD pages with almost zero boilerplate.
:link: https://grails.apache.org/guides/grails-fields-custom-widgets-and-wrappers/8/guide/index.html

Who’s got a guide idea burning a hole in their brain? We’d love to see what topics the community wants to tackle next! :rocket:

**Comment by carlosorrego** [comment]
Scaffolding for MongoDB is buggy.
I do not know how common is MongoDB+grails combo. But I love it.

**Comment by jeffscottbrown** [comment]
> It was a shame when at the time of the release of Grails 3 there was plans to remove this feature from Grails 

I am not saying that didn't happen, but I don't remember that.  Did we publish something to that effect at one point?

**Comment by kico-itexto** [comment]
Hi Jeff,

yes: at the time the first version of 3.0 did not have the scaffolding
feature. It was added on the next release.

Em qua., 2 de out. de 2024 às 15:10, Jeff Scott Brown <
***@***.***> escreveu:

> It was a shame when at the time of the release of Grails 3 there was plans
> to remove this feature from Grails
>
> I am not saying that didn't happen, but I don't remember that. Did we
> publish something to that effect at one point?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orgs/grails/discussions/13533#discussioncomment-10824004>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC6W7AH73LLLQTGBI4OXAY3ZZQZI5AVCNFSM6AAAAABKJONN7CVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTAOBSGQYDANA>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Henrique Lobo Weissmann
Diretor Técnico
<https://www.facebook.com/itexto> <https://www.twitter.com/itexto>
<https://www.linkedin.com/company/itexto>
<https://www.instagram.com/itextobr>
*Tel:* +55 (31) 2573-1979 <+55(31)2573-1979>
*Email:* ***@***.***
*Web:* www.itexto.com.br

**Comment by jeffscottbrown** [comment]
> at the time the first version of 3.0 did not have the scaffolding
> feature. It was added on the next release.

I honestly forgot that happened.  Thank you!

**Comment by jamesfredley** [comment]
https://grails.apache.org/docs/snapshot/guide/scaffolding.html - current docs 

https://github.com/apache/grails-core/edit/8.0.x/grails-doc/src/en/guide/scaffolding.adoc

**Comment by kico-itexto** [comment]
And it's an awesome article! Thank you!

Em qui., 28 de mai. de 2026 às 22:57, James Fredley <
***@***.***> escreveu:

>
> https://grails.apache.org/guides/grails-fields-custom-widgets-and-wrappers/8/guide/index.html
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/apache/grails-core/discussions/13533?email_source=notifications&email_token=AC6W7AFJWXCD46NQHWJQQDT45DVAZA5CNFSNUABIM5UWIORPF5TWS5BNNB2WEL2ENFZWG5LTONUW63SDN5WW2ZLOOQXTCNZQHE4TINRTUZZGKYLTN5XKMYLVORUG64VFMV3GK3TUVRTG633UMVZF6Y3MNFRWW#discussioncomment-17099463>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AC6W7AAPBDRJABJPBTKOUAD45DVAZAVCNFSM6AAAAACY5J7C3KVHI2DSMVQWIX3LMV43URDJONRXK43TNFXW4Q3PNVWWK3TUHMYTOMBZHE2DMMY>
> .
> Triage notifications, keep track of coding agent tasks and review pull
> requests on the go with GitHub Mobile for iOS
> <https://github.com/notifications/mobile/ios/AC6W7AAWWUWZTK3SBQJH3JL45DVAZA5CNFSNUABIM5UWIORPF5TWS5BNNB2WEL2ENFZWG5LTONUW63SDN5WW2ZLOOQXTCNZQHE4TINRTUZZGKYLTN5XKMYLVORUG64VFMV3GK3TUVJTG633UMVZF62LPOM>
> and Android
> <https://github.com/notifications/mobile/android/AC6W7AF2OCVEUTZ3F6LLEBD45DVAZA5CNFSNUABIM5UWIORPF5TWS5BNNB2WEL2ENFZWG5LTONUW63SDN5WW2ZLOOQXTCNZQHE4TINRTUZZGKYLTN5XKMYLVORUG64VFMV3GK3TUVZTG633UMVZF6YLOMRZG62LE>.
> Download it today!
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>


-- 

Henrique Lobo Weissmann
Diretor Técnico


*Tel:* +55 (31) 2573-1979 <+55(31)2573-1979>
*Email:* ***@***.***
*Web:* www.itexto.com.br
